<?xml version="1.0"?>
<launch>
    <!-- Include the rsp_lab4 launch file -->
    <include file="$(find-pkg-share mirte_gazebo)/launch/rsp_lab4.launch.xml">
        <arg name="gui" value="true"/>
    </include>

    <!-- Launch the PCL obstacle detector node -->
    <node pkg="pcl_obstacle_detector" exec="pcl_obstacle_detector" name="pcl_obstacle_detector_node" output="screen"/>

    <!-- Launch the control barrel world nodes -->
    <node pkg="control_barrel_world" exec="control_barrel_world" name="control_barrel_world_node" output="screen">
        <param name="publish_interval_ms" value="100"/>
        <param name="pedestrian_area_thresh" value="2500.0"/>  <!-- Must be a float! -->
        <param name="linear_speed" value="0.1"/>
        <param name="angular_speed" value="0.3"/>
    </node>
</launch>
